File: lesson-23/Python/credit_card_customers.py
from pyspark.sql import SparkSession
from pyspark.ml import PipelineModel
from pyspark.sql.functions import *
import sys

if __name__ == "__main__":
    if len(sys.argv) != 4:
        print("Usage: CreditCardCustomers <path-to-model> <path-to-input> <path-to-output>")
        sys.exit(-1)

    spark = SparkSession \
        .builder \
        .master("local[*]")\
        .appName("CreditCardCustomers") \
        .getOrCreate()

    model = PipelineModel.load(sys.argv[1])

    data = spark \
        .read \
        .option("header", "true") \
        .option("inferSchema", "true") \
        .csv(sys.argv[2])

    predicted = model.transform(data)

    predicted \
        .filter(col("prediction") == 1) \
        .select("CLIENTNUM") \
        .repartition(1) \
        .write \
        .mode("overwrite") \
        .csv(sys.argv[3])

    spark.stop()

File: lesson-21/Scala/src/main/scala/ru/otus/sparkml/CreditCardCustomers.scala
package ru.otus.sparkml

import org.apache.spark.sql.{SaveMode, SparkSession}
import org.apache.spark.ml.PipelineModel

object CreditCardCustomers {
  def main(args: Array[String]): Unit = {
    if (args.length != 3) {
      println("Usage: SparkML <path-to-model> <path-to-input> <path-to-output>")
      sys.exit(-1)
    }

    val spark = SparkSession.builder
      .appName("CreditCardCustomers")
      .config("spark.sql.debug.maxToStringFields", 100)
      .getOrCreate()

    import spark.implicits._

    try {
      val model = PipelineModel.load(args(0))

      val data = spark.read
        .option("header", "true")
        .option("inferSchema", "true")
        .csv(args(1))

      val prediction = model.transform(data)

      prediction
        .filter($"prediction" === 1)
        .select("CLIENTNUM")
        .repartition(1)
        .write
        .mode(SaveMode.Overwrite)
        .csv(args(2))

    } catch {
      case e: Exception =>
        println(s"ERROR: ${e.getMessage}")
        sys.exit(-1)
    } finally {
      spark.stop()
    }
  }
}
